modules/augment/src/augmenter.cpp

```diff
@@ -0,0 +1,88 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "precomp.hpp"
+namespace cv { namespace augment {
+
+Augmenter::Augmenter() {}
+
+void Augmenter::add(Ptr<Transform> transformation, float prob)
+{
+    transformations.push_back(transformation);
+    probs.push_back(prob);
+}
+
+void Augmenter::applyImages(const std::vector<Mat>& imgs, OutputArrayOfArrays dstImgs)
+{
+    dstImgs.create(imgs.size(), 1, 0, -1, true);
+    RNG rng;
+
+    for (size_t i = 0; i < imgs.size(); i++)
+    {
+        Mat img = imgs[i].clone();
+
+            for (size_t j = 0; j < transformations.size(); j++)
+            {
+                float prob = rng.uniform(0.f, 1.f);
+
+                if (prob <= probs[j])
+                {
+                    transformations[j]->init(imgs[i]);
+                    transformations[j]->image(img, img);
+                }
+
+            }
+
+            dstImgs.create(img.size(), img.type(), i, true);
+            Mat dstImg = dstImgs.getMat(i);
+            img.copyTo(dstImg);
+
+    }
+
+}
+
+
+void Augmenter::applyImagesWithMasks(const std::vector<Mat>& imgs,
+    const std::vector<Mat>& masks,
+    OutputArrayOfArrays dstImgs,
+    OutputArrayOfArrays dstMasks)
+{
+    dstImgs.create(imgs.size(), 1, 0, -1, true);
+    dstMasks.create(masks.size(), 1, 0, -1, true);
```

Comment:
Please check existing usage of `OutputArrayOfArrays`, in example `imagesFromBlob` in https://github.com/opencv/opencv/blob/master/modules/dnn/src/dnn.cpp.

Reply:
in this function he stacked the images using cv::merge which depends on all the matrices having the same dimensions.. which is not the case for us because the Augmenter class supports input images of any different resolutions.. so it can only be used this way acrually.